src/RestWrite.js

```diff
@@ -802,7 +802,7 @@ RestWrite.prototype.createSessionTokenIfNeeded = function() {
   if (this.className !== '_User') {
     return;
   }
-  if (this.query) {
+  if (this.query && !this.data.authData) {
```

Comment:
A little logic here

query yes auth yes (link existing user with auth generate)
query no auth yes (new user with auth generate)
query no auth no (new user generate)
query yes auth no (update user no auth don't generate)

Reply:
can you please put this useful tidbit of info a comment right there or wherever you think it might be most helpful to anyone ever poking around in here again?

Reply:
Done